ykcompile/src/lib.rs

```diff
@@ -137,19 +229,31 @@ impl TraceCompiler {
         // Set available registers to R11-R8, RDX, RCX
         let regs = vec![11, 10, 9, 8, 2, 1];
         let assembler = dynasmrt::x64::Assembler::new().unwrap();
+
         let mut tc = TraceCompiler {
             asm: assembler,
             available_regs: regs,
             assigned_regs: HashMap::new(),
         };
+
+        let mut err = None;
         for i in 0..tt.len() {
-            let t = tt.op(i);
-            match t {
+            let res = match tt.op(i) {
                 TirOp::Statement(st) => tc.statement(st),
-                TirOp::Guard(_) => tc.nop(), // FIXME Implement guards.
+                TirOp::Guard(g) => tc.guard(g),
+            };
+
+            if let Err(e) = res {
+                err = Some(e);
```

Comment:
Why don't we call `tc.crash_dump()` here so we don't need to assign the `err` variable.

Reply:
Fixed.